File: application/app/src/main/java/com/yahyeet/boardbook/activity/account/IAccountFragmentHolder.java
package com.yahyeet.boardbook.activity.account;

/**
 * Binds events from fragments to the activity so that the fragments does not depend on the parents presenter.
 */
public interface IAccountFragmentHolder {


	/**
	 * Method tries to log into app given valid parameters
	 *
	 * @param email    of the account trying to log in
	 * @param password of the account trying to log in
	 */
	void loginAccount(String email, String password);

	/**
	 * Method tries to create a new account from parameters and log in with it
	 *
	 * @param email    of the new account
	 * @param password of the new account
	 * @param username of the new account
	 */
	void registerAccount(String email, String password, String username);
}

File: application/app/src/main/java/com/yahyeet/boardbook/activity/account/IAccountFragmentHolder.kt
package com.yahyeet.boardbook.activity.account

interface IAccountFragmentHolder {


    /**
     * Method tries to log into app given valid parameters
     *
     * @param email    of the account trying to log in
     * @param password of the account trying to log in
     */
    fun loginAccount(email: String, password: String)

    /**
     * Method tries to create a new account from parameters and log in with it
     *
     * @param email    of the new account
     * @param password of the new account
     * @param username of the new account
     */
    fun registerAccount(email: String, password: String, username: String)
}
